### PR TITLE
[no ticket][risk=no] CircleCI Names and anchors for common steps

### PR DESCRIPTION
**Pull request by jaycarlton**
New CircleCI UI makes heavy use of the optional `name` attribute on job steps. I've added more, and tweaked a few existing ones for clarity. I condensed a couple of common steps into YAML anchors. There is support for more first-class reuse with parameterization in v2.1. This [ticket ](https://precisionmedicineinitiative.atlassian.net/browse/RW-4645) covers taking advantage of the new features and migrating to 2.1.

Check CircleCI run for new titles.
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -154,25 +155,15 @@ jobs:
...
+      - run: *ensure_branch_has_api_changes
       - restore_cache:
           keys:
           - api-integration-cache-{{ checksum "~/workbench/api/build.gradle" }}
-          - api-integration-cache-
-      - run:
-          working_directory: ~/workbench
-          # Used to call gsutil from the circle environment.
-          command: ci/activate_creds.sh api/circle-sa-key.json
+          - api-integration-cache-`
```
Comment (calbach) [inline .circleci/config.yml:163]: was this backtick intentional?

File: .circleci/config.yml [hunk 2]
```diff
@@ -295,21 +275,22 @@ jobs:
     steps:
       - checkout
       - run:
-          name: Fetch Submodules
+          name: Update Git Submodules
```
Comment (calbach) [inline .circleci/config.yml:278]: anchor?

File: .circleci/config.yml (hunk 1 above)
Comment (jaycarlton) [reply .circleci/config.yml]: No, good catch. I'm apparently too hard on Mac keyboards and they've started rebelling on me.

File: .circleci/config.yml (hunk 2 above)
Comment (jaycarlton) [reply .circleci/config.yml]: yep

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
-    TERM: dumb
-
+anchors:
+  defaults: &defaults
+    docker:
+      - image: allofustest/workbench:buildimage-0.0.16
+    working_directory: ~/workbench
+  ensure_branch_has_api_changes: &ensure_branch_has_api_changes
+    name: Ensure Branch is on Origin
+    #Circle is bad about setting up your remotes for you, you must use origin/
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply .circleci/config.yml]: Yeah, trying to understand where the remotes came in distracted me from the actual code.

**Comment by jaycarlton** [comment]
@aweng98 I looked briefly a #3440 and I think this can go in safely either before or after yours. Thanks for taking on the 2.1 upgrade.

**Comment by jaycarlton** [comment]
@calbach for some reason, the steps for [this job](https://app.circleci.com/pipelines/github/all-of-us/workbench/176/workflows/8abb7b8f-05c2-4455-b092-9a71eb8e786a/jobs/78071) were listed in the wrong order in the UI.

Also, is there a reason we don't have `store_test_results` set so we can use the test result UI?

**Comment by calbach** [comment]
RE wrong order: looks correct to me, what is out of order?

RE store_test_results: nope, I don't think anyone has tried

**Comment by jaycarlton** [comment]
I think the ordering was an artifact of skipping the test job when no changes matched. It looked like a couple of other steps (e.g. submodules) were run unnecessarily after (or in parallel to) that. I'll keep an eye out.

**Pull request merged**